Gems/ImGui/Code/Source/ImGuiManager.cpp

```diff
@@ -729,10 +715,19 @@ void ImGuiManager::ToggleThroughImGuiVisibleState()
         case DisplayState::Visible:
             m_clientMenuBarState = DisplayState::Hidden;
 
-            // Restore old cursor state
-            AzFramework::InputSystemCursorRequestBus::Event(AzFramework::InputDeviceMouse::Id,
-                &AzFramework::InputSystemCursorRequests::SetSystemCursorState,
-                m_previousSystemCursorState);
+            const bool inGame = !gEnv->IsEditor() || gEnv->IsEditorGameMode(); 
```

Comment:
Suggestion: Move the definition of `inGame` outside of the switch statement, then in the DisplayState::Hidden case, you can set `m_previousSystemCursorStateInGame = inGame;`. This would make the logic a little bit easier to read because it provides explicit continuity between how the variable is set and how it's being checked.